homeassistant/components/light/vera.py

```diff
@@ -0,0 +1,181 @@
+running on 80
+Example: http://192.168.1.21:3480/
+
+
+device_data
+*Optional
+This contains an array additional device info for your Vera devices.  It is not
+required and if not specified all lights configured in your Vera controller
+will be added with default values.
+
+
+These are the variables for the device_data array:
+
+vera_id
+*Required
+The Vera device id you wish these configuration options to be applied to
+
+
+name
+*Optional
+This parameter allows you to override the name of your Vera device in the HA
+interface, if not specified the value configured for the device in your Vera
+will be used
+
+
+exclude
+*Optional
+This parameter allows you to exclude the specified device from homeassistant,
+it should be set to "true" if you want this device excluded
+
+"""
+import logging
+import time
+
+from homeassistant.helpers import ToggleDevice
+# pylint: disable=no-name-in-module, import-error
+import homeassistant.external.vera.vera as veraApi
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """ Find and return Vera lights. """
+    try:
+        base_url = config.get('vera_controller_url')
+        if not base_url:
+            _LOGGER.error(
+                "The required parameter 'vera_controller_url'"
+                " was not found in config"
+            )
+            return False
+
+        device_data = config.get('device_data', None)
+
+        controller = veraApi.VeraController(base_url)
+        devices = controller.get_devices('Switch')
+
+        lights = []
+        for device in devices:
+            extra_data = get_extra_device_data(device_data, device.deviceId)
+            exclude = False
+            if extra_data:
+                exclude = extra_data.get('exclude', False)
+
+            if exclude is not True:
+                lights.append(VeraLight(device, extra_data))
+
+        add_devices_callback(lights)
+    # pylint: disable=broad-except
+    except Exception as inst:
+        _LOGGER.error("Could not find Vera lights: %s", inst)
+        return False
+
+
+def get_extra_device_data(device_data, device_id):
+    """ Gets the additional configuration data by Vera device Id """
+    if not device_data:
+        return None
+
+    for item in device_data:
+        if item.get('vera_id') == device_id:
+            return item
+
+    return None
+
+
+class VeraLight(ToggleDevice):
+    """ Represents a Vera light """
+    is_on_status = False
+    # for debouncing status check after command is sent
+    last_command_send = 0
+    extra_data = None
+
+    def __init__(self, vera_device, extra_data=None):
+        self.vera_device = vera_device
+        self.extra_data = extra_data
+
+    @property
+    def name(self):
+        """ Get the mame of the light. """
+        if self.extra_data and self.extra_data.get('name'):
```

Comment:
This logic will result in the same name every time. It can be moved to the constructor and saved in `_name`. That way you only have to return `_name` in this method.